lms/djangoapps/gating/tests/test_api.py

```diff
@@ -34,143 +37,214 @@ def setUp(self):
             category='sequential',
-            display_name='untitled sequential 1'
+            display_name='gating sequential 1',
+            graded=True,
+            format='Homework',
         )
         self.seq2 = ItemFactory.create(
             parent_location=self.chapter1.location,
             category='sequential',
-            display_name='untitled sequential 2'
+            display_name='gated sequential 2',
+            graded=True,
+            format='Homework',
         )
-
-        # create vertical
-        self.vert1 = ItemFactory.create(
-            parent_location=self.seq1.location,
-            category='vertical',
-            display_name='untitled vertical 1'
+        self.seq3 = ItemFactory.create(
+            parent_location=self.chapter1.location,
+            category='sequential',
+            display_name='sequential 3',
+            graded=True,
+            format='Homework',
         )
 
         # create problem
-        self.prob1 = ItemFactory.create(
-            parent_location=self.vert1.location,
+        self.gating_prob1 = ItemFactory.create(
+            parent_location=self.seq1.location,
+            category='problem',
+            display_name='gating problem 1',
+        )
+        self.gated_prob2 = ItemFactory.create(
+            parent_location=self.seq2.location,
+            category='problem',
+            display_name='gated problem 2',
+        )
+        self.prob3 = ItemFactory.create(
+            parent_location=self.seq3.location,
             category='problem',
-            display_name='untitled problem 1'
+            display_name='problem 3',
         )
 
         # create orphan
-        self.prob2 = ItemFactory.create(
+        self.orphan = ItemFactory.create(
             parent_location=self.course.location,
             category='problem',
-            display_name='untitled problem 2'
+            display_name='orphan'
+        )
+
+        self.prereq_milestone = None
+
+    def setup_gating_milestone(self, min_score):
+        """
+        Setup a gating milestone for testing.
+        Gating content: seq1 (must be fulfilled before access to seq2)
+        Gated content: seq2 (requires completion of seq1 before access)
+        """
+        gating_api.add_prerequisite(self.course.id, self.seq1.location)
+        gating_api.set_required_content(self.course.id, self.seq2.location, self.seq1.location, min_score)
+        self.prereq_milestone = gating_api.get_gating_milestone(self.course.id, self.seq1.location, 'fulfills')
+
+    def verify_access_to_gated_content(self, user, expected_access):
+        """
+        Verifies access to gated content for the given user is as expected.
+        """
+        # clear the request cache to flush any cached access results
+        RequestCache.clear_request_cache()
+
+        # access to gating content (seq1) remains constant
+        self.assertTrue(bool(has_access(user, 'load', self.seq1, self.course.id)))
```

Comment:
`bool()` is redundant here. `assertTrue(x)` already evaluates to `assert bool(x) is True`: See  https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue